src/components/Popover/hooks.ts

```diff
@@ -29,7 +32,11 @@ export const useDisableScroll = (disableScrollRef, isDisabled) => {
  * @param {Boolean} isActive - if the popover is currently open/active
  * @param {Boolean} closeCallback - function that closes the popover
  */
-export const useCloseOnScroll = (closeOnScrollRef, isActive, closeCallback) => {
+export const useCloseOnScroll = (
+  closeOnScrollRef: MutableRefObject<HTMLElement> | undefined,
+  isActive: boolean,
+  closeCallback: () => void
+): void => {
```

Comment:
We're going to need to pair so you can teach me these weird React generics 😂

Reply:
sure thing, it has a bit of a steep learning curve if you're not used to Generics in general but after that it's not complex at all.